lib/Dialect/SystemC/SystemCOps.cpp

```diff
@@ -325,6 +326,90 @@ LogicalResult SCFuncOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// InstanceDeclOp
+//===----------------------------------------------------------------------===//
+
+void InstanceDeclOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
+  setNameFn(getInstanceHandle(), getName());
+}
+
+Operation *InstanceDeclOp::getReferencedModule(const hw::HWSymbolCache *cache) {
+  if (cache)
+    if (auto *result = cache->getDefinition(getModuleNameAttr()))
+      return result;
+
+  auto topLevelModuleOp = (*this)->getParentOfType<ModuleOp>();
+  return topLevelModuleOp.lookupSymbol(getModuleName());
+}
+
+Operation *InstanceDeclOp::getReferencedModule() {
+  return getReferencedModule(/*cache=*/nullptr);
+}
+
+LogicalResult
+InstanceDeclOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
```

Comment:
Thanks for adding this exhaustive checking and including checks for each code path in the lit tests.